### PR TITLE
GH-11 Add Deb package generation support

### PR DESCRIPTION
**Pull request by rfernandes**
Update required CMakeLists boost version
Add Debian package generation.
Running cpack -GDEB yields a debian package:

```
dpkg -I dist/toolbox-cpp-snapshot-x86_64.deb
 new debian package, version 2.0.
 size 8005316 bytes: control archive=2299 bytes.
     274 bytes,    11 lines      control
    5200 bytes,    69 lines      md5sums
 Architecture: amd64
 Depends: boost (>=1.67)
 Description: Toolbox - The Reactive C++ Toolbox
 Homepage: https://www.reactivemarkets.com
 Maintainer: development@reactivemarkets.com
 Package: toolbox-cpp
 Priority: optional
 Section: devel
 Version: snapshot
 Installed-Size: 23438
```

### CODE REVIEW

**Comment by rfernandes** [comment]
Packages can be created directly on ArchLinux by installing [AUR - DPKG](https://aur.archlinux.org/packages/dpkg/)

File: CMakeLists.txt [hunk 1]
```diff
@@ -191,9 +191,9 @@ add_subdirectory(image)
...
 add_subdirectory(example)
 
-set(CPACK_GENERATOR "TGZ;RPM")
+set(CPACK_GENERATOR "TGZ;RPM;DEB")
 
-SET(CPACK_OUTPUT_FILE_PREFIX dist)
+set(CPACK_OUTPUT_FILE_PREFIX dist)
 set(CPACK_PACKAGE_CONTACT "development@reactivemarkets.com")
 set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Toolbox - The Reactive C++ Toolbox")
 set(CPACK_PACKAGE_HOMEPAGE_URL "https://www.reactivemarkets.com")
```
Comment (marayl) [inline CMakeLists.txt]: Perhaps  we should change this to "https://github.com/reactivemarkets/toolbox-cpp" while we're here?

File: CPackProjectConfig.cmake [hunk 2]
```diff
@@ -1,7 +1,10 @@
 message(STATUS "CMAKE_INSTALL_PREFIX: ${CMAKE_INSTALL_PREFIX}")
 if("${CPACK_GENERATOR}" MATCHES "TGZ")
   set(CPACK_PACKAGING_INSTALL_PREFIX "/toolbox")
-else("${CPACK_GENERATOR}" MATCHES "RPM")
+elseif("${CPACK_GENERATOR}" MATCHES "DEB")
+  set(CPACK_DEBIAN_PACKAGE_DEPENDS "boost (>=1.67)")
```
Comment (marayl) [inline CPackProjectConfig.cmake]: Is it possible to add a similar dependency for the RPM generator?

**Comment by marayl** [comment]
This PR fixes #16.

**Pull request merged**